test/runnable/test_cdvecfill.d

```diff
@@ -123,730 +123,830 @@ alias AliasSeq(Args...) = Args;
 // dfmt off
 alias baselineCases = AliasSeq!(
     Code!(ubyte, 16 / ubyte.sizeof)([
-        /* push   rax                     */ 0x50,
+        /* push   rbp                     */ 0x55,
+        /* mov    rbp,rsp                 */ 0x48, 0x8b, 0xec,
         /* movd   xmm0,edi                */ 0x66, 0x0f, 0x6e, 0xc7,
         /* punpcklbw xmm0,xmm0            */ 0x66, 0x0f, 0x60, 0xc0,
         /* punpcklwd xmm0,xmm0            */ 0x66, 0x0f, 0x61, 0xc0,
         /* pshufd xmm0,xmm0,0x0           */ 0x66, 0x0f, 0x70, 0xc0, 0x00,
-        /* pop    rcx                     */ 0x59,
+        /* pop    rbp                     */ 0x5d,
         /* ret                            */ 0xc3,
     ]),
```

Comment:
Why was this code change necessary?

Reply:
Apparently when emitting position-independent code, different registers are used. There's an update script included in this file which I used to update the test cases.

Reply:
Would it be prudent to keep the original test for Non-PIC and simply keep this PR as a simple addition?

Reply:
Yes I could use `version(D_PIC)`, but this would make the update script more complex. And I'm not sure it's necessary, because this script is only run under Linux `x86_64` atm and since 2.072.2 dmd uses `-fPIC` for compiled programs on Linux as default (for Darwin `-fPIC` is even fixed). Anyhow, if I add the old test cases, this would only make sense, if they are run as part of the test suite and then if not in `version(PIE)` we would have to detect if the environment is hardened and then avoid running the test cases.
And in case this will ever be needed, the original code is still saved in git and alternatively it can be easily regenerated with Martin's fancy update script.
So I would only do this extra work if  @MartinNowak or @WalterBright think it's essential.